### ISSUE TITLE
autorestic returning no data when running cron

### ISSUE THREAD

**Issue by katchy3132** [cupcakearmy/autorestic#248]
### Discussed in https://github.com/cupcakearmy/autorestic/discussions/244

<div type='discussions-op-text'>

<sup>Originally posted by **katchy3132** October  7, 2022</sup>
I have a cronjob setup (running as root ) to run autorestic once an hour : 

`33 * * * *   date >> /var/log/autorestic.log &&  /usr/local/bin/autorestic -c /opt/backups/autorestic.yml --ci cron >>  /var/log/autorestic.log 2>&1`

when it comes time to run it seems the restic command just stops which triggers a failure ( I have deleted a bunch of the --excludes for brevity ) : 

```
Fri 07 Oct 2022 02:33:01 
Using config:    /opt/backups/autorestic.yml
Using lock:      /opt/backups/.autorestic.lock.yml
Skipping "main", not due yet.
Fri 07 Oct 2022 03:33:01 
Using config:    /opt/backups/autorestic.yml
Using lock:      /opt/backups/.autorestic.lock.yml


    Backing up location "main"

Backend: remote-b2
> Executing: restic backup --exclude *.tmp  --exclude /root/.cache --exclude .zcompdump --tag ar:cron --tag ar:location:main /etc /home /root /opt/backups


Running hooks
> curl https://xyz.com/api/push/xxxxxxxx?status=down&msg=FAIL&ping=
> Executing: /bin/bash -c curl https://xyz.com/api/push/xxxxxx?status=down&msg=FAIL&ping=
{"ok":true} 
```

but if I run the backup command directly everything goes fine: 
`sudo /usr/local/bin/autorestic -c /opt/backups/autorestic.yml backup -a`

this is the cron line from the config : 
`cron: 30 03 * * *
` 

any ideas what could be happening ? 
thanks!

BTW two feature requests: 
- add a timestamp to the -ci so that logging is easier ( I add it in the cronjob ) 
- add ability to bump up the restic logging level without using "autorestic exec"
</div>

**Comment by patrick-motard** [comment]
Same issue here. Autorestic works fine when run directly in the shell. The backup succeeds but then the failure hook is called. No output explaining why it failed. I've tried:

```
/usr/local/bin/autorestic -vvv --ci cron >> /tmp/autorestic.log 2>&1
```

The logs just say which locations were backed up. The backblaze job succeeds, then it gets to the "Running hooks" section and the fail hook runs.

**Comment by mill1000** [comment]
Also seeing this. Autorestic seems to think the cronjob is executed correctly, as subsequent calls indicate "not due yet", but there are no new snapshots in the repo.

**Comment by mill1000** [comment]
Ok, resolved my problem. Restic wasn't in my path when executed via cron.

The documentation does mention the path setup here: https://autorestic.vercel.app/location/cron

My `/etc/crontab` had the path setup but I'm guessing that isn't inherited by files in `/etc/cron.d/`

**Comment by katchy3132** [comment]
Indeed adding the PATH mentioned here solved this.  
> The documentation does mention the path setup here: https://autorestic.vercel.app/location/cron#crontab

thanks! and closing this.

**Comment by mill1000** [comment]
I do think Autorestic should be handling this in a more graceful manner. Being unable to find restic should be an obvious reported error, not a silent fail.

**Comment by katchy3132** [comment]
> I do think Autorestic should be handling this in a more graceful manner. Being unable to find restic should be an obvious reported error, not a silent fail.

yes absolutely . ill reopen this and hopefully @cupcakearmy  can add this at some point.
